cfn-resources/project-ip-access-list/cmd/resource/resource.go

```diff
@@ -146,16 +181,56 @@ func Delete(req handler.Request, prevModel *Model, currentModel *Model) (handler
 // List handles the List event from the Cloudformation service.
 // NO-OP
 func List(req handler.Request, prevModel *Model, currentModel *Model) (handler.ProgressEvent, error) {
-	setup()
-	log.Debugf("Got list request - returning read - %v", currentModel)
-	readEvent, err := Read(req, prevModel, currentModel)
-	log.Debugf("List readEvent:+%v   --------------------------- error:%+v", readEvent, err)
-	if readEvent.OperationStatus == handler.Failed {
-		return readEvent, nil
+	modelValidation := validateModel(ListRequiredFields, currentModel)
+	if modelValidation != nil {
+		return *modelValidation, nil
+	}
+
+	client, err := util.CreateMongoDBClient(*currentModel.ApiKeys.PublicKey, *currentModel.ApiKeys.PrivateKey)
+	if err != nil {
+		return handler.ProgressEvent{
+			OperationStatus:  handler.Failed,
+			Message:          err.Error(),
+			HandlerErrorCode: cloudformation.HandlerErrorCodeInvalidRequest}, nil
 	}
+
+	var pageNum, itemsPerPage int
+	var includeCount bool
+
+	if currentModel.ListOptions != nil {
+		if currentModel.ListOptions.PageNum != nil {
+			pageNum = *currentModel.ListOptions.PageNum
+		}
+		if currentModel.ListOptions.IncludeCount != nil {
+			includeCount = *currentModel.ListOptions.IncludeCount
+		}
+		if currentModel.ListOptions.ItemsPerPage != nil {
+			itemsPerPage = *currentModel.ListOptions.ItemsPerPage
+		}
+	}
+
+	listOptions := &mongodbatlas.ListOptions{
+		PageNum:      pageNum,
+		IncludeCount: includeCount,
+		ItemsPerPage: itemsPerPage,
+	}
```

Comment:
no need to check for nil, if `pageNum` (or the other props) is set to null, the endpoint is not going to use it 
```suggestion
	if currentModel.ListOptions != nil {
	    listOptions := &mongodbatlas.ListOptions{
		PageNum:      *currentModel.ListOptions.PageNum,
		IncludeCount: *currentModel.ListOptions.IncludeCount,
		ItemsPerPage: *currentModel.ListOptions.ItemsPerPage,
	   }
	}


```